packages/grafana-data/src/types/query.ts

```diff
@@ -8,11 +10,18 @@ export enum DataTopic {
 }
 
 /**
- * In 8.2, this will become an interface
- *
  * @public
  */
-export type DatasourceRef = string;
+export interface DatasourceRef {
+  /** The plugin type-id */
+  type?: string;
+
+  /** Specific datasource instance */
+  uid?: string;
+
+  /** The schema version for query target JSON  */
+  modelVersion?: ModelVersion;
```

Comment:
we are not using this yet, but it is important think though where this should be saved